LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/RadioMetadata.cs

```diff
@@ -3,35 +3,7 @@
 
 namespace LoRaWan.NetworkServer.BasicsStation
 {
-    public class RadioMetadata
-    {
-        public DataRateIndex DataRate { get; }
-        public Hertz Frequency { get; }
-        public RadioMetadataUpInfo UpInfo { get; }
+    public record RadioMetadata(DataRateIndex DataRate, Hertz Frequency, RadioMetadataUpInfo UpInfo);
```

Comment:
@atifaziz could you tell me it this codeQL warning is appropriate? am I missusing records?

Reply:
Sorry, I don't see the warning. Where should I be looking?

Reply:
I believe those are false positives and can be ignored. With positional records, you are declaring the primary constructor with arguments at the same time as the properties of the record and CodeQL is complaining that your constructor arguments are named the same as the properties and thus shadowing them. There is by-design of the language.